src/rgw/rgw_rados.cc

```diff
@@ -1137,6 +1137,12 @@ int RGWRados::register_to_service_map(const string& daemon_type, const map<strin
     name = name.substr(4);
   }
   int ret = rados.service_daemon_register(daemon_type, name, metadata);
```

Comment:
Sorry, I didn't completely understand this part, do we want to register twice with two different names?

Reply:
No this will just be overridden if we enter the condition.

Reply:
but why register twice?